pyxem/utils/segment_utils.py

```diff
@@ -0,0 +1,313 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+from numpy.ma import masked_where
+
+import matplotlib.pyplot as plt
+
+from scipy.ndimage import distance_transform_edt, label, binary_erosion
+from scipy.spatial import distance_matrix
+from scipy.signal import convolve2d
+
+from skimage.feature import peak_local_max
+from skimage.filters import sobel, threshold_li
+from skimage.morphology import watershed, disk
+
+from sklearn.cluster import DBSCAN
+
+import warnings
+
+
+def norm_cross_corr(image, template):
+    """Calculates the normalised cross-correlation between an image
```

Comment:
@pc494 do we not have a normalized xc function somewhere else for peak finding?

Reply:
The xc method uses skimage.feature.match_template

Reply:
The peakfinder is cross correlation, this might be better described as "image pmcc" - anyhow, the core line should probably make use of https://docs.scipy.org/doc/numpy/reference/generated/numpy.corrcoef.html

Reply:
yeah this certainly doesn't feel like something we should be implementing...

Reply:
Will update code to reflect this discussion